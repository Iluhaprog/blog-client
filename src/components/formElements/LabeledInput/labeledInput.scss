@import '../../../vars.scss';

.form__box {
    max-width: 320px;
    width: 100%;
    font-family: $main-font;
    font-size: 14px;
    box-shadow: $shadow;
    border-radius: 5px;
    margin-bottom: 20px;
    border: $border-2;

    label {
        display: block;
        padding: 13px 20px;
        min-width: 110px;
        text-align: center;
        border-right: $border-2;
        color: $text-color-2;
        transition: .2s ease-in-out;
    }

    input {
        border: none;
        width: 100%;
        border-radius: 5px;
        padding: 10px;
        font-size: 16px;

        &:focus {
            outline: none;
            & + label {
                color: $text-color-3;
                min-width: 95px;
            }
        }
    }

}