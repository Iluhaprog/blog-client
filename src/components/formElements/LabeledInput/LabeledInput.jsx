import { Field } from "redux-form";
import './labeledInput.scss';

export default props => (
    <div className='form__box row-reverse'>
        <Field 
            type={props.type}
            name={props.name}
            component={props.component}
        />
        <label htmlFor={props.name}>{props.label}</label>
    </div>
);