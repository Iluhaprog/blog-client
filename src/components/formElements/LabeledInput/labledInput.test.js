import React from 'react';
import { render } from '@testing-library/react';
import { reduxForm } from 'redux-form';
import { Provider } from 'react-redux';
import LabeledInput from './LabeledInput';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();


const decor = el => {
    const form = () => (<form>{el}</form>);
    return reduxForm({
        form: 'test',
    })(form);
}

test('Test labled input view', () => {
    const store  = mockStore({});
    const Lbin = decor(
        <LabeledInput 
            type='text'
            name='test'
            component='input'
            label='TEST'
        />
    );
    const { asFragment } = render(<Provider store={store}><Lbin /></Provider>);
    expect(asFragment()).toMatchSnapshot();
});