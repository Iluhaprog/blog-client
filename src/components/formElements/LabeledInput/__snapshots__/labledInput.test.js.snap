// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test labled input view 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form__box row-reverse"
    >
      <input
        name="test"
        type="text"
        value=""
      />
      <label
        for="test"
      >
        TEST
      </label>
    </div>
  </form>
</DocumentFragment>
`;
